Backend/CalendarService.cs

```diff
@@ -0,0 +1,77 @@
+using FRITeam.Swapify.Backend.Interfaces;
+using FRITeam.Swapify.SwapifyBase.Entities;
+using FRITeam.Swapify.SwapifyBase.Settings;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FRITeam.Swapify.Backend
+{
+    public class CalendarService : ICalendarService
+    {
+        private readonly int dayOfWeek = 0;
+        private readonly CalendarSettings _calendarSettings;
+        private readonly DateTime startDateTime;
+        private readonly DateTime endDateTime;
+
+        public CalendarService(IOptions<CalendarSettings> calendarSettings)
+        {
+            _calendarSettings = calendarSettings.Value;
+
+            DateTime dateNow = DateTime.Now;
+            if (dateNow.CompareTo(_calendarSettings.StartWinter) == 1 && dateNow.CompareTo(_calendarSettings.EndWinter) == -1)
+            {
+                startDateTime = _calendarSettings.StartWinter;
+                endDateTime = _calendarSettings.EndWinter;
+            }
+            else
+            {
+                startDateTime = _calendarSettings.StartSummer;
+                endDateTime = _calendarSettings.EndSummer;
+            }
+            dayOfWeek = dateNow.DayOfWeek == DayOfWeek.Sunday ? 7 : (int)dateNow.DayOfWeek;
```

Comment:
I do not know the CalendarService lifecycle. It is created as singleton, so is used by all user requests?
If yes, it can live more then a day, is `dayOfWeek` initialization OK?

Reply:
Yes it is singleton so initialization was not ok. In latest commit I moved logic from constructor to function and problem should be solved.